### PR TITLE
API: Add `finch.random` function

### PR DESCRIPTION
**Pull request by mtsokol**
Hi @willow-ahrens @hameerabbasi!

This small PR adds `finch.random` to mimic `sparse.random` (required in benchmarks that I'm running).

### CODE REVIEW

File: src/finch/tensor.py [hunk 1]
```diff
@@ -396,7 +396,11 @@ def construct_csf(cls, arg: TupleOf3Arrays, shape: tuple[int, ...]) -> "Tensor":
         return Tensor(cls.construct_csf_jl_object(arg, shape))
 
 
-def fsprand(*args):
+def random(shape, density=0.01, random_state=None):
+    args = [*shape, density]
+    if random_state is not None:
+        rng = jl.default_rng(random_state)
+        args = [rng] + args
```
Comment (hameerabbasi) [inline src/finch/tensor.py]: Here we should handle the case that `random_state` is a `np.random.Generator` instance, in which case we can just generate some bytes from it to pass to `jl.default_rng`.

Comment (mtsokol) [reply src/finch/tensor.py]: Done! `jl.default_rng` only accepts an integer.

Comment (willow-ahrens) [reply src/finch/tensor.py]: looks good!

File: src/finch/tensor.py [hunk 2]
```diff
@@ -396,7 +396,15 @@ def construct_csf(cls, arg: TupleOf3Arrays, shape: tuple[int, ...]) -> "Tensor":
         return Tensor(cls.construct_csf_jl_object(arg, shape))
 
 
-def fsprand(*args):
+def random(shape, density=0.01, random_state=None):
+    args = [*shape, density]
+    if random_state is not None:
+        if isinstance(random_state, np.random.Generator):
+            seed = random_state.integers(np.iinfo(np.int32).max)
```
Comment (hameerabbasi) [inline src/finch/tensor.py:403]: Would a `np.uint64` work here?

**Review by hameerabbasi — APPROVED** [review]
LGTM, besides the one comment.

File: src/finch/tensor.py (hunk 2 above)
Comment (mtsokol) [reply src/finch/tensor.py:403]: Sure it can! I will change that later.

**Pull request merged**